generators/E906LegacyGen/SQPrimaryVertexGen.C

```diff
@@ -16,34 +16,39 @@ from Kun to E1039 experiment in Fun4All framework
 
 #include "SQPrimaryVertexGen.h"
 //#include "SQBeamlineObject.h"
-
-SQPrimaryVertexGen::SQPrimaryVertexGen() 
+double beam_vtx_x; 
+double beam_vtx_y;
+ TF2* beam_global;
+SQPrimaryVertexGen::SQPrimaryVertexGen():
+ _beam_profile(nullptr)
 {
 inited = false;
-
-}
-SQPrimaryVertexGen::~SQPrimaryVertexGen(){}
-
-void SQPrimaryVertexGen::Initfile(){
+ // beam_vtx_x = 0;
+ // beam_vtx_y = 0;
  
 }
 
-void SQPrimaryVertexGen::InitRun(PHCompositeNode* topNode)
+SQPrimaryVertexGen::~SQPrimaryVertexGen()
 {
-  beam_profile= false;
-  beamProfile = NULL;
-  beamProfile = new TF2("beamProfile", "exp(-0.5*(x-[0])*(x-[0])/[1]/[1])*exp(-0.5*(y-[2])*(y-[2])/[3]/[3])", -10., 10., -10., 10.);
-  beamProfile->SetParameter(0, 0.0);
-  beamProfile->SetParameter(1, 0.68);
-  beamProfile->SetParameter(2, 0.0);
-  beamProfile->SetParameter(3, 0.76);
```

Comment:
As I see these lines were removed, you seem to stop using "beamProfile".If so, the variable "beamProfile" itself should be removed from this class.

Reply:
I have used it in the generateVtxPerp() function.
https://github.com/abinashpun/e1039-core/blob/c357feb85eb1fbb188cd5c3b078b7b2852a8ee28/generators/E906LegacyGen/SQPrimaryVertexGen.C#L289-L292

Reply:
I thought "beamProfile" was always "null" (without "= new TF2") and thus "if(beamProfile)" is always "false".  But now I see that you moved this part to "Fun4Sim.C".

What is an advantage of new'ing "beam_profile" in Fun4Sim.C?  I think we better keep such technical object inside the class.

Reply:
I came up with this idea because I think we don't know the exact beam profile. So, in the future, we can set it simply via Fun4Sim without digging up the e1039-core area. Please let me know what you think about it.

Reply:
Yes, the parameters should be adjustable.  I think a simpler member function can do that such as "SQPrimaryVertexGen::setProfile(x_center, y_center, x_width, y_width)".

I expect we won't change the functional shape itself (i.e. Gaussian).  Thus user need not handle TF2 directly.

Reply:
Ok. But I think we also expect to have some tail function in addition to Gaussian. I can try as you suggested. But for now, I came up with this convention. 
https://github.com/E1039-Collaboration/e1039-analysis/blob/558faecc40a62186f62db8310af1206ef46e7949/SimChainDev/Fun4Sim.C#L75-L91

Let me know what you think about it.

Reply:
So I would agree to keep the current interface, as you expect soon(?) to change the function shape.  I thought such change was rare.